### PR TITLE
fix: load IdsTokenValidationServiceExtension properly

### LINKED ISSUE #1274: ids-token-validation ServiceExtension not registered properly
# Bug Report

## Describe the Bug

The [ids-token-validation ServiceExtension service locator file](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/data-protocols/ids/ids-token-validation/src/main/resources/META-INF.services/org.eclipse.dataspaceconnector.spi.system.ServiceExtension
) is placed at the wrong path - should be [data-protocols/ids/ids-token-validation/src/main/resources/META-INF/services/](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/data-protocols/ids/ids-token-validation/src/main/resources/META-INF/services/org.eclipse.dataspaceconnector.spi.system.ServiceExtension) instead.

### Expected Behavior

The [IdsTokenValidationServiceExtension](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/data-protocols/ids/ids-token-validation/src/main/java/org/eclipse/dataspaceconnector/ids/token/validation/IdsTokenValidationServiceExtension.java) should be picked up, initialized and started properly.

### Observed Behavior

[IdsTokenValidationServiceExtension](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/data-protocols/ids/ids-token-validation/src/main/java/org/eclipse/dataspaceconnector/ids/token/validation/IdsTokenValidationServiceExtension.java) does not load

## Steps to Reproduce

* Start connector with ids-token-validation extension added to the classpath

## Context Information

- Used version is [EDC @508b59142e0915285a27936fa399a87fcc4bd79b](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/commit/)

## Detailed Description
_If applicable, add screenshots and logs to help explain your problem._

## Possible Implementation

- rename the directory from `META-INF.services` to `META-INF/services`

<sub>Denis Neuling <denis.neuling@mercedes-benz.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

### PR DESCRIPTION
**Pull request by juliapampus**
## What this PR changes/adds

Move `ServiceExtension` of `IdsTokenValidationServiceExtension` from `META-INF.services` to `META-INF/services`. 

Exclude `extensions:ids:ids-token-validation` where `:extensions:iam:iam-mock` is used - for samples and tests.

## Why it does that

This caused reported issues during initialization.

## Further notes

--

## Linked Issue(s)

Closes #1274 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Review by ronjaquensel — APPROVED** [review]

**Comment by juliapampus** [comment]
@ndr-brt The `ids-token-validation` is loaded within `:data-protocols:ids` in the basic launcher. The problem is that this module needs `:extensions:iam:oauth2:oauth2-spi` as a dependency. I am wondering why the basic launcher is using `:extensions:iam:iam-mock` and how we could solve this without causing failing test in the pipeline. WDYT?

File: extensions/http-provisioner/build.gradle.kts [hunk 1]
```diff
@@ -28,6 +28,7 @@ dependencies {
     api(project(":spi:transfer-spi"))
     api(project(":spi:web-spi"))
     implementation(project(":extensions:api:auth-spi"))
+    implementation(project(":extensions:api:auth-tokenbased"))
```
Comment (ndr-brt) [inline extensions/http-provisioner/build.gradle.kts:31]: I'm not sure this is a good idea, in this way we'll have the http-provisioned forced to use the token based auth.
I would provide a deafult pass through authentication service with the `@Provider(isDefault = true)` annotation on the http webhook extension

**Review by ndr-brt — COMMENTED** [review]
apart the comment is all good for me, guess those `exclude` would disappear as soon as we'll cleanup the `ids` module

File: extensions/http-provisioner/build.gradle.kts (hunk 1 above)
Comment (juliapampus) [reply extensions/http-provisioner/build.gradle.kts:31]: I will create a new issue for this, so we get this PR closed, then you can take a look next week

Comment (juliapampus) [reply extensions/http-provisioner/build.gradle.kts]: #1358

**Pull request merged**